src/serviceClients/catalogClient.ts

```diff
@@ -24,6 +24,8 @@ export class CatalogClient extends HttpClient {
 
   @withSpanAsyncV4
   public async exists(productId: string, productType: string): Promise<boolean> {
+    const activeSpan = trace.getActiveSpan();
+    activeSpan?.updateName('catalogClient.exists');
     const res = await this.findByProductIdAndType(productId, productType);
     return res.length > 0;
   }
```

Comment:
add for both of the find functions spans

Reply:
done